File: Khelmianov-Artem/Term4/Block3/Dekanat/src/main/kotlin/examsystem/ExamSystem.kt
package examsystem

typealias StudentId = Long
typealias CourseId = Long
typealias Credit = Pair<StudentId, CourseId>

interface ExamSystem {
    val count: Int
    fun add(studentId: StudentId, courseId: CourseId)
    fun remove(studentId: StudentId, courseId: CourseId)
    fun contains(studentId: StudentId, courseId: CourseId): Boolean
}

File: Ilya-Dudnikov/term_4/block_3/task_1/ExamSystem/ExamSystem.cs
namespace ExamSystem;

public class ExamSystem : IExamSystem
{
    private ISet<Credit> set;

    public ExamSystem(ISet<Credit> set)
    {
        this.set = set;
    }
    
    public void Add(long studentId, long courseId) => set.Add(new Credit(studentId, courseId));

    public void Remove(long studentId, long courseId) => set.Remove(new Credit(studentId, courseId));

    public bool Contains(long studentId, long courseId) => set.Contains(new Credit(studentId, courseId));

    public int Count => set.Count;
}